Lib/logging/__init__.py

```diff
@@ -1558,33 +1563,31 @@ def findCaller(self, stack_info=False, stacklevel=1):
         f = currentframe()
         #On some versions of IronPython, currentframe() returns None if
         #IronPython isn't run with -X:Frames.
-        if f is not None:
-            f = f.f_back
-        orig_f = f
-        while f and stacklevel > 1:
-            f = f.f_back
-            stacklevel -= 1
-        if not f:
-            f = orig_f
-        rv = "(unknown file)", 0, "(unknown function)", None
-        while hasattr(f, "f_code"):
```

Comment:
Note that the new code doesn't verify that frame object contain an `f_code` attribute. It does verify that the frame object is not `None`. The `f_code` attribute is prescribed by the CPython data model and the `warnings` module doesn't check for it's presence either (it does check for stack exhaustion/`None`, like the new code here).